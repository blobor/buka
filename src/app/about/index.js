import React from 'react'

const About = () => {
  return (
    <article>
      <h1>About!!</h1>
    </article>
  )
}

export default About