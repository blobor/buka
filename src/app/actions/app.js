import * as actionTypes from './actionTypes'

const toggleNavigationDrawer = () => {
  return {
    type: actionTypes.TOGGLE_NAVIGATION_DRAWER
  }
}

const closeNavigationDrawer = () => {
  return {
    type: actionTypes.CLOSE_NAVIGATION_DRAWER
  }
}

export {
  closeNavigationDrawer,
  toggleNavigationDrawer
}
