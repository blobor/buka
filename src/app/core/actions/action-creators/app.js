import {
  TOGGLE_DEVELOPER_MODE
} from '../action-types'

const toggleDeveloperMode = developerMode => {
  return {
    type: TOGGLE_DEVELOPER_MODE,
    developerMode
  }
}

export {
  toggleDeveloperMode
}