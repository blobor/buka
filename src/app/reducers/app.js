import { version } from '../../../package.json'

const initialState = {
  name: 'Buka',
  version: version
}

export default function (state = initialState, action) {
  switch (action.type) {
    default:
      return state
  }
}
