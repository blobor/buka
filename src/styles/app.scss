@import "styles.scss";
@import "lifts-table.scss";
@import "skipass-info.scss";
