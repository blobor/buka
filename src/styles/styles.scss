@import "constant.scss";
@import "screen-sizes.scss";

html, body {
    font-family: 'Roboto', sans-serif;
}

.buka {
    display: flex;
    flex-direction: column;
    font-size: 14px;
    min-height: 100vh;

  &__container {
      display: flex;
      flex: 1 0 auto;
      flex-direction: column;
      align-items: center;
      box-sizing: border-box;
      padding: 0 10pt;
      margin: $header-height auto calc(#{$footer-height} + 5pt);
      width: 100%;
      max-width: 980px;
  }

  &__footer {
    flex-shrink: 0;
    position: fixed;
    left: 0;
    bottom: 0;
    width: 100%;
    height: $footer-height;
    background-color: #FFF;
  }
}

@media (max-width: $xsmall) {
  .buka__container {
    padding: 0 2pt;
  }
}

@media (max-width: $small) {
  .buka__container {
    padding: 0 5pt;
  }
}

.buka-skipass-number__form {
    width: 100%;
    display: flex;
    align-items: center;
    align-self: center;
    justify-content: center;
    flex-wrap: wrap;
}

.buka-skipass-number__input input {

  &::-ms-clear {
    display: none;
  }

  &:invalid {
    box-shadow: none;
  }

  &::-webkit-search-cancel-button {
    -webkit-appearance: none;
  }
}

.buka-skipass-search__add-button--hidden {
  display: none !important;
}

.bottom-navigation {
  display: flex;
  justify-content: center;

  &__link {
    max-width: 168px;
    flex: 1;

    button {
      // workaround for FF
      width: 100%;
    }
  }
}
