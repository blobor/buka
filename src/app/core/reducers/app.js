import { fromJS } from 'immutable'
import { TOGGLE_DEVELOPER_MODE } from '../actions/action-types'
import { appName, version, development } from '../../../config.js'

const initialState = fromJS({
  name: appName,
  developerMode: development,
  version: version
})

export default (state = initialState, action) => {
  switch (action.type) {
    case TOGGLE_DEVELOPER_MODE:
      return state.set('developerMode', action.developerMode)
    default:
      return state
  }
}
