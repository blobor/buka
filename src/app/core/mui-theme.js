import { fullWhite } from 'material-ui/styles/colors'
import { getMuiTheme as getTheme } from 'material-ui/styles'

export const getMuiTheme = userAgent => {
  return getTheme({
    userAgent: userAgent,
    textField: {
      hintColor: '#565656'
    },
    svgIcon: {
      color: fullWhite
    }
  })
}