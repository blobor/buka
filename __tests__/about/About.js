/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import { MuiThemeProvider } from 'material-ui/styles'
import { getMuiTheme } from 'app/core/mui-theme'
import { About } from 'app/about'

describe('About', () => {
  it('should renders correctly', () => {
    const muiTheme = getMuiTheme()
    const mockProps = {
      appVersion: '0.0.1'
    }

    const tree = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <About {...mockProps} />
      </MuiThemeProvider>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
