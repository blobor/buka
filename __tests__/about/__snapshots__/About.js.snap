exports[`About should renders correctly 1`] = `
<article
  className="about__container">
  <div
    style={
      Object {
        "muiPrepared": true,
        "padding": "8px 0px 8px 0px",
      }
    }>
    <div>
      <span
        disabled={undefined}
        href={undefined}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
        onTouchTap={[Function]}
        open={null}
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
            "background": "none",
            "backgroundColor": null,
            "border": 10,
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "cursor": "pointer",
            "display": "block",
            "fontFamily": "Roboto, sans-serif",
            "fontSize": 16,
            "fontWeight": "inherit",
            "lineHeight": "16px",
            "margin": 0,
            "muiPrepared": true,
            "outline": "none",
            "padding": 0,
            "position": "relative",
            "textDecoration": "none",
            "transform": "translate(0, 0)",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "verticalAlign": null,
          }
        }
        tabIndex={0}>
        <div
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}>
          <div
            style={
              Object {
                "marginLeft": 0,
                "muiPrepared": true,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
                "position": "relative",
              }
            }>
            <div
              style={
                Object {
                  "muiPrepared": true,
                }
              }>
              Version
            </div>
          </div>
        </div>
      </span>
    </div>
    <hr
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "border": "none",
          "height": 1,
          "margin": 0,
          "marginLeft": 0,
          "marginTop": -1,
          "muiPrepared": true,
        }
      } />
  </div>
</article>
`;
